### DISCUSSION TITLE
Kill a stuck job.

### DISCUSSION THREAD

**Discussion by GoOnYahGoodThing** [quartznet/quartznet#1590] [Q&A]
Hi,
I've been getting to grips with interrupting jobs etc.
And its all very straightforward... as long as the job checks the cancellation token.

But I'm just wondering what can be done if a job gets stuck or hung and needs to just be Killed .

Is there a way to do that?
This is just a 'nuclear option' that I'm sure my boss is doing to want to know about as some of the jobs we currently run manually can get stuck and require killing the process.. it doesn't happen often but.. 

Will shutting down the app kill all the jobs that are running maybe?

Thanks

**Comment by lahma** [comment]
I don't think there's an easy option for that. Such things usually require running on separate process to allow such nuclear option. Generally such jobs should be fixed to observe custom timeouts and cancellation.

**Reply by GoOnYahGoodThing** [reply]
Yeah, I would obviously intend to write any jobs to watch the cancellation token etc.
But sh1t happens.. :-) and jobs can fall victim to  locks on database tables and other external things can have effects too.
But I suppose if I just shut down the app it should kill all the sub-processes..
